### ISSUE TITLE
Allow audio and pdf upload

### ISSUE THREAD

**Issue by hidrarga** [tootsuite/mastodon#7495]
Currently, it is only possible to upload videos and images on Mastodon. In [this discussion](https://discourse.joinmastodon.org/t/sharing-audio-content-as-multimedia/325), Maloki explains that audio uploads are left out to avoid copyright issues. However, this argument doesn't hold as this should be to the admins and moderators of the instance to take responsibility of copyright issues (as it is already the case for texts, videos and images). At the very least, allowing audio uploads should be an option for each instance's admins.

I would like to add to this request the possibility to upload pdfs. As far as I know, they are not more concerned by copyright infringement than images, let's say.

Thanks :)

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by mal0ki** [comment]
I'm just going to reference to this issue: https://github.com/tootsuite/mastodon/issues/6987
Which is requesting admins be allowed to pick and choose.

**Comment by trwnh** [comment]
I have to say that "copyright" is not really a reason to not implement audio posts, especially considering that
1) mastodon allows image and video upload, as well as text
2) copyright claims should be directed at instance admins
3) copyright should not be such a consideration in the first place for free software, nor should it be facilitated or given technological concessions so as to become enshrined or reified within that software.

re: #6987 configurable types are not enough if there isn't a handler built into mastodon for Audio or Document objects federated over ActivityPub.

**Comment by Gargron** [comment]
@trwnh Actually, it is a very valid reason. Kodi is a Linux desktop environment that works like a media player. There is nothing illegal about it, it does not prescribe you to watch pirated movies. And yet, Google decided to block it from its search results on piracy grounds: <https://torrentfreak.com/google-adds-kodi-to-autocomplete-piracy-filter-180328/>

As-is, Mastodon is not a file sharing service. Even though it allows people to attach images and videos to toots, there is no expectation of format or quality fidelity (e.g. max 8MB videos, webm/mp4 only, jpg/png only, large images are downsized to at most 1280^2 pixels). So even though you could encode a song in a video, it's far from sharing MP3 or FLAC files. You can't share ZIP archives. You can't send someone a PSD file.

I think it was a good decision to not allow audio uploads. Yes maybe it would've been fine. Or maybe the media would have branded Mastodon as a pirated music sharing service, and our project would start getting blocked from search results. Not a risk I am interested in taking *at all* when the most simple solution is to just include a link to soundcloud, instaud.io, vocaroo.com, dropbox, google cloud, your own server, or whatever else. Mastodon should not be a file sharing service!! It's a specialized problem domain with specialized solutions.

TL;DR this issue is a duplicate of #4827

**Comment by trwnh** [comment]
This seems more like some kind of double standard. Tumblr, for example, allows audio uploads. Vocaroo, Instaudio, SoundCloud, etc. are built around audio uploads. No one would primarily describe any of the above as a "pirated music sharing service". Kodi's case was due to heavy-handedness from Google.

Mastodon may not be a file sharing service, and I personally would not advocate for it to be one either -- file uploads are definitely better suited to file hosting sites. But audio is a fundamental type of communication, like language or images.

I'm fine with moving discussion of this over to #4827, though.

**Comment by trwnh** [comment]
@nightpool theoretically the same could be said for videos. Or images. Or text. There's no limit to how deep that copyright rabbit-hole goes. :/

**Comment by nightpool** [comment]
yes but audio uploads are different in culture and social context, and I'm
saying that tumblr is a bad justification to use because they've had lots
of problems with them

**Comment by nightpool** [comment]
(and not video, pictures, etc)

**Comment by 320x200** [comment]
Agreed, this limitation seems arbitrary and the parallel with copyright issues is sketchy, given the kind of appropriated material is already widely circulating on the fediverse. There are a lot of musicians and performers waiting to share and post snippets of their work, or simply release some tracks under free/permissive license, or whatever conditions they see fit, and they have to add a video track to their audio to be able to share it, which is quite convoluted.

Please consider allowing the upload of audio files and provide an <audio> element for their playback.

Thanks!
